optimisation/ScanSelectionsTree.py

```diff
@@ -192,6 +191,9 @@
 cSignifVsRest, hSignifVsRest, cEstimVsCut, hEstimVsCut = [], [], [], []
 counter = 0
 for iPt, (ptMin, ptMax) in enumerate(zip(ptMins, ptMaxs)):
+    # reshuffle bkg and take only a fraction of it, seed fixed for reproducibility
+    dfBkg = dfBkg.sample(frac=fractionstokeep[iPt], random_state=42).reset_index(drop=True)
+
```

Comment:
Hi @danielbattistini, this looks like a possible bug to me. Here you are sampling and modifying the `bkg` dataframe integrated in pt, so from the second bin onward you are sampling from the "reduced" dataframe obtained before.
Moreover, but this was a problem probably already present, I think it is better to first select the pt range of your interest and then do the sampling. The difference is minor if your input file contains data from a small pt range but when you consider all the analysis pt range you could have problems.

Reply:
Thank you for notifying me! That's right, it should be fixed now.